quickwit/quickwit-serve/src/index_api/rest_handler.rs

```diff
@@ -500,16 +504,43 @@ fn delete_index_handler(
         ("index_id" = String, Path, description = "The index ID to delete."),
     )
 )]
-/// Deletes index.
+/// Deletes indexes from patterns
 async fn delete_index(
-    index_id: String,
+    index_id_patterns: Vec<String>,
     delete_index_query_param: DeleteIndexQueryParam,
     index_service: IndexService,
 ) -> Result<Vec<SplitInfo>, IndexServiceError> {
-    info!(index_id = %index_id, dry_run = delete_index_query_param.dry_run, "delete-index");
-    index_service
-        .delete_index(&index_id, delete_index_query_param.dry_run)
-        .await
+    info!(index_id_patterns = ?index_id_patterns, dry_run = delete_index_query_param.dry_run, "delete_index");
+    // get all index_ids by index_id_patterns
+    let indexes_metadata = index_service
```

Comment:
can you move the code in the `IndexService`?

Ideally, the rest handler should be very dumb:

```rust
let split_info: Vec<SplitInfo> = index_service.delete_indexes(index_id_patterns).await?;
```

Reply:
@fmassot Thanks for the suggestion
I think it would be better to move the logic to IndexService, and keep the rest handler as simple as possible.

Reply:
yes exactly